src/ol/uri.js

```diff
@@ -0,0 +1,17 @@
+goog.provide('ol.uri');
+
+
+/**
+ * Appends query parameters to a URI.
+ *
+ * @param {string} uri The original URI, which may already have query data.
+ * @param {!Object} params An object where keys are URI-encoded parameter keys,
+ *     and the values are arbitrary types or arrays.
+ * @return {string} The new URI.
+ */
+ol.uri.appendParams = function(uri, params) {
+  var qs = Object.keys(params).map(function(k) {
+    return encodeURIComponent(k) + '=' + encodeURIComponent(params[k]);
+  });
+  return uri + (uri.indexOf('?') === -1 ? '?' : '&') + qs.join('&');
```

Comment:
for example, http://www.icc.cat/eng/Home-ICC/Digital-geoinformation/Online-services-Geoservices/Quick-WMS-WMTS-of-raster-cartography includes the '?' in the base url. http://www.ign.es/wms-inspire/mapa-raster?request=GetCapabilities&service=WMS has a parameter in the base url and finishes with '&'. I've found this sort of thing to be quite common with WMS. Of course, you could get around this by specifying in the docs that any trailing '?' or '&' should be left out, but ISTM better if the library can handle this. Having said that, it's probably also the case that browsers are clever enough to ignore duplicates like this.